src/device-state/target-state.ts

```diff
@@ -85,27 +94,29 @@ export const update = async (
 		const endpoint = url.resolve(apiEndpoint, `/device/v2/${uuid}/state`);
 		const request = await getRequestInstance();
 
-		const params: Headers = {
+		const params: CoreOptions = {
 			json: true,
+			headers: {
+				Authorization: `Bearer ${deviceApiKey}`,
+				'If-None-Match':
+					typeof cache?.etag === 'string' ? cache.etag : undefined,
```

Comment:
This should actually be fine as
```suggestion
					cache?.etag,
```
since it can only be a string if it exists and so the end result is the same